src/state/account.ts

```diff
@@ -329,24 +320,20 @@ export const accountSlice = createSlice({
     viewedOrders: (state) => {
       state.hasUnseenOrderUpdates = false;
     },
-    setBalances: (state, action: PayloadAction<Record<string, AccountBalance>>) => {
-      state.balances = action.payload;
-    },
-    setStakingBalances: (state, action: PayloadAction<Record<string, AccountBalance>>) => {
-      state.stakingBalances = action.payload;
-    },
-    setStakingDelegations: (state, action: PayloadAction<StakingDelegation[]>) => {
-      state.stakingDelegations = action.payload;
-    },
-    setUnbondingDelegations: (state, action: PayloadAction<UnbondingDelegation[]>) => {
-      state.unbondingDelegations = action.payload;
-    },
-    setStakingRewards: (state, action: PayloadAction<StakingRewards>) => {
-      state.stakingRewards = action.payload;
-    },
-    setTradingRewards: (state, action: PayloadAction<TradingRewards>) => {
-      state.tradingRewards = action.payload;
-    },
+    ...generateTypedSetterActions(
```

Comment:
conceptually very cool but 😢 unfortunately i find the old code more readable. 

this makes sense once you understand what `generateTypedSetterActions` does but it takes a bit of time to onboard to and is a bit more mental overhead to remember.

i may be the minority in this though - i typically like to avoid abstractions